app/assets/javascripts/discourse/app/components/quote-button.js

```diff
@@ -184,100 +215,98 @@ export default Component.extend(KeyEnterEscape, {
     // on Desktop, shows the button at the beginning of the selection
     // on Mobile, shows the button at the end of the selection
     const isMobileDevice = this.site.isMobileDevice;
-    const { isIOS, isAndroid, isSafari, isOpera } = this.capabilities;
+    const { isIOS, isAndroid, isOpera } = this.capabilities;
     const showAtEnd = isMobileDevice || isIOS || isAndroid || isOpera;
 
-    // Don't mess with the original range as it results in weird behaviours
-    // where certain browsers will deselect the selection
-    const clone = firstRange.cloneRange();
-
-    // create a marker element containing a single invisible character
-    const markerElement = document.createElement("span");
-    markerElement.appendChild(document.createTextNode("\ufeff"));
-
-    // on mobile, collapse the range at the end of the selection
-    if (showAtEnd) {
-      clone.collapse();
-    }
-    // insert the marker
-    clone.insertNode(markerElement);
-
-    // retrieve the position of the marker
-    const $markerElement = $(markerElement);
-    const markerOffset = $markerElement.offset();
-    const parentScrollLeft = $markerElement.parent().scrollLeft();
-    const $quoteButton = $(this.element);
-
-    // remove the marker
-    const parent = markerElement.parentNode;
-    parent.removeChild(markerElement);
-    // merge back all text nodes so they don't get messed up
-    parent.normalize();
-
-    // work around Safari that would sometimes lose the selection
-    if (isSafari) {
-      this._reselected = true;
-      selection.removeAllRanges();
-      selection.addRange(clone);
-    }
+    const boundaryPosition = getRangeBoundaryRect(
+      firstRange,
+      showAtEnd ? true : false
```

Comment:
```suggestion
      showAtEnd
```

Or is one of those helpers returning a non-bool? 🙂

Reply:
Fixed :+1: